internal/services/storage/storage_share_resource_test.go

```diff
@@ -185,6 +186,35 @@ func TestAccStorageShare_largeQuota(t *testing.T) {
 	})
 }
 
+func TestAccStorageShare_accessTier(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_share", "test")
+	r := StorageShareResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.accessTier(data, string(shares.CoolAccessTier)),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.accessTier(data, string(shares.HotAccessTier)),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.accessTier(data, string(shares.TransactionOptimizedAccessTier)),
```

Comment:
can we hard-code the string values for these? that protects against the value changing casing etc